extensions/ql-vscode/src/databases/github-database-updates.ts

```diff
@@ -0,0 +1,205 @@
+import { CodeqlDatabase } from "./github-database-api";
+import { DatabaseItem, DatabaseManager } from "./local-databases";
+import { Octokit } from "@octokit/rest";
+import { CodeQLCliServer } from "../codeql-cli/cli";
+import { AppCommandManager } from "../common/commands";
+import { getLanguageDisplayName } from "../common/query-language";
+import { showNeverAskAgainDialog } from "../common/vscode/dialog";
+import { downloadGitHubDatabaseFromUrl } from "./database-fetcher";
+import { withProgress } from "../common/vscode/progress";
+import { window } from "vscode";
+import { GitHubDatabaseConfig } from "../config";
+import { joinLanguages, promptForDatabases } from "./github-database-download";
+
+export type DatabaseUpdate = {
+  database: CodeqlDatabase;
+  databaseItem: DatabaseItem;
+};
+
+type DatabaseUpdateStatusUpdateAvailable = {
+  type: "updateAvailable";
+  databaseUpdates: DatabaseUpdate[];
+};
+
+type DatabaseUpdateStatusUpToDate = {
+  type: "upToDate";
+};
+
+type DatabaseUpdateStatusNoDatabase = {
+  type: "noDatabase";
+};
+
+type DatabaseUpdateStatus =
+  | DatabaseUpdateStatusUpdateAvailable
+  | DatabaseUpdateStatusUpToDate
+  | DatabaseUpdateStatusNoDatabase;
+
+export function isNewerDatabaseAvailable(
+  databases: CodeqlDatabase[],
+  owner: string,
+  name: string,
+  databaseManager: DatabaseManager,
+): DatabaseUpdateStatus {
+  // Sorted by date added ascending
+  const existingDatabasesForRepository = databaseManager.databaseItems
+    .filter(
+      (db) =>
+        db.origin?.type === "github" &&
+        db.origin.repository === `${owner}/${name}`,
+    )
+    .sort((a, b) => (a.dateAdded ?? 0) - (b.dateAdded ?? 0));
+
+  if (existingDatabasesForRepository.length === 0) {
+    return {
+      type: "noDatabase",
+    };
+  }
+
+  // Sort order is guaranteed by the sort call above. The newest database is the last one.
+  const newestExistingDatabasesByLanguage = new Map<string, DatabaseItem>();
+  for (const existingDatabase of existingDatabasesForRepository) {
+    newestExistingDatabasesByLanguage.set(
+      existingDatabase.language,
+      existingDatabase,
+    );
+  }
+
+  const databaseUpdates = Array.from(newestExistingDatabasesByLanguage.values())
+    .map((newestExistingDatabase): DatabaseUpdate | null => {
+      const origin = newestExistingDatabase.origin;
+      if (origin?.type !== "github") {
+        return null;
+      }
+
+      const matchingDatabase = databases.find(
+        (db) => db.language === newestExistingDatabase.language,
+      );
+      if (!matchingDatabase) {
+        return null;
+      }
+
+      if (matchingDatabase.commit_oid === origin.commitOid) {
+        return null;
+      }
+
+      return {
```

Comment:
Yes, I think that is something we should consider. I've now changed the check to also consider the database's `created_at` date. This can result in more prompts (e.g. if there is a scheduled workflow run to scan `main` every day and there are no commits), but I think that's fine for now; we can always reconsider.